src/brasil/gov/vcge/upgrades/v2001/__init__.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding:utf-8 -*-
+from brasil.gov.vcge.dx.interfaces import IVCGEDx
+from brasil.gov.vcge.logger import logger
+from plone import api
+
+import csv
+import os
+
+
+INFO_URL = 'https://www.governoeletronico.gov.br/documentos-e-arquivos/VCGE_DEPARA_01.pdf'
+
+
+class Migrator():
+    """Context manager to handle term migration."""
+
+    def __init__(self):
+        path = os.path.dirname(__file__)
+        self.filename = os.path.join(path, 'VCGE_DEPARA_01.csv')
+
+    def __enter__(self):
+        """Read equivalences file and create dictionary."""
+        self.equivalences = {}
+        logger.debug('Reading equivalence file')
+        with open(self.filename) as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                new = row['termo_vcge_2B']
+                if new:
+                    old = row['termo_vcge_1']
+                    self.equivalences[old] = new
+
+    def __exit__(self, *args):
+        pass  # update the catalog?
+
+    def migrate(self, obj):
+        """Migrate existing terms on objects."""
+        if obj is None:
+            return  # invalid object
+
+        if not obj.skos:
+            return  # nothing to do (empty or None)
+
+        new = ()
+        # ignore terms with no equivalence
+        for term in obj.skos:
+            if term in self.equivalences:
+                new += self.equivalences[term]
```

Comment:
In this case all terms would be migrated? or are there any example of term that would remain as it is from old to new version?

for example, we have 3 old terms ['term1', 'term2', 'term3'];
'term2' should be replaced by 'term4' and 'term3' by 'term5';

with the code written now, the result would be ['term4', 'term5'] because 'term1' is not into ``equivalences`` dict.

Is it fine like this?

Reply:
check the tests.